### PR TITLE
uniffi-bindgen: 0.17.0 -> 0.19.3

### PR DESCRIPTION
**Pull request by vtuan10**
###### Description of changes
upstream update
changes: https://github.com/mozilla/uniffi-rs/blob/main/CHANGELOG.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veehaitch** [comment]
Result of `nixpkgs-review pr 182193` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>uniffi-bindgen</li>
  </ul>
</details>

File: pkgs/development/tools/uniffi-bindgen/update.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -p nix cargo rsync nix-update
+#! nix-shell -i bash
+
+set -euo pipefail
+
+if [[ -z "${UPDATE_NIX_ATTR_PATH+x}" ]]; then
+    echo "Error: run the following command from nixpkgs root:" >&2
+    echo "  nix-shell maintainers/scripts/update.nix --argstr package <package name>" >&2
```
Comment (winterqt) [inline pkgs/development/tools/uniffi-bindgen/update.sh:9]:
```suggestion
    echo "  nix-shell maintainers/scripts/update.nix --argstr package uniffi-bindgen" >&2
```

**Review by winterqt — CHANGES_REQUESTED** [review]

**Comment by winterqt** [comment]
Please squash that change into the commit that added the update script.

**Review by winterqt — APPROVED** [review]

**Comment by bobby285271** [comment]
Looks like the update script does not work (#190648) because of https://github.com/Mic92/nix-update/issues/80#issuecomment-1237875667
Log: https://r.ryantm.com/log/uniffi-bindgen/2022-09-10.log

**Pull request merged**